### ISSUE TITLE
Enlarge camera streams when objects are detected

### ISSUE THREAD

**Issue by tyzbit** [blakeblackshear/frigate#12887]
**Describe what you are trying to accomplish and why in non technical terms**
I want to be able to see all camera feeds at the same time for full observability but I also want to be able to see with high detail any events where objects are detected live.

**Describe the solution you'd like**
I want the home screen to show all of the camera feeds at once and at the same size by default. When a camera detects an object, I want that camera feed to enlarge on the home screen. If multiple cameras detect an object, those cameras are enlarged but share space between each other.

**Describe alternatives you've considered**
The new feature to resize cameras in camera groups provides a great solution to let users arrange their cameras into a static arrangement, but it must be manually set up and there is no automatic resizing in response to objects being detected.

**Additional context**
Example:

![image](https://github.com/user-attachments/assets/ea665df8-db3c-4e8e-9c91-8d350eff6d23)

Here, cameras are the same size because no objects are being detected.

![image](https://github.com/user-attachments/assets/bbe607fe-167b-4931-ba30-ac9d6b82f87e)

Imagining that a person was detected in the driveway camera, it would resize automatically and the remaining cameras would be pushed to the side and resized if necessary.

I think this would help people with a large number of cameras effectively monitor them but be able to see detail without viewing an alert or clicking through the dashboard. 

I know it could be challenging to implement due to deciding how to handle different camera aspect ratios, different web browser display ratios, multiple events at once, ordering of the cameras on the dashboard, efficiently using the space, etc.

**Comment by hawkeye217** [comment]
This is how [birdseye](https://docs.frigate.video/configuration/birdseye) works currently, and it can be configured with all sorts of . I don't think we're looking to implement those kinds of features with HTML in a browser.

**Comment by NickM-27** [comment]
That's what the red outline is for, the camera can be clicked to be opened

**Comment by tyzbit** [comment]
> This is how [birdseye](https://docs.frigate.video/configuration/birdseye) works currently

Not quite, it removes the streams without detection after a time. It doesn't really resize the cameras, it enlarges them by removing other cameras. This feature request envisions all cameras continuing to be visible.

> That's what the red outline is for, the camera can be clicked to be opened
Yes, but as I said "I think this would help people with a large number of cameras effectively monitor them but be able to see detail without viewing an alert or clicking through the dashboard". If you click one camera, you no longer see the others. Resizing them in one view allows you to keep total awareness.

**Comment by hawkeye217** [comment]
There are too many variables with numbe rof c

> Not quite, it removes the streams without detection after a time. It doesn't really resize the cameras, it enlarges them by removing other cameras. This feature request envisions all cameras continuing to be visible.

There are too many variables between camera setups, aspect ratios, streaming protocols, and browser versions to realistically accomplish this in a way that is supportable for all users.

Birdseye is Frigate's best solution for viewing active cameras simultaneously.
